### PR TITLE
SeaMoth/Prawn Suit/Cyclops  Colours & Name persistence

### PR DESCRIPTION
**Pull request by ASimpson87**
Cyclops needs more work.

### CODE REVIEW

**Comment by NitroxPower** [comment]
Good job :+1: does this also fix the bug that upgrade modules dont get saved  when they are on a vehicle?

**Comment by ASimpson87** [comment]
> Good job 👍 does this also fix the bug that upgrade modules dont get saved when they are on a vehicle?

This doesn't look at upgrade modules , however it's on my to do list.

File: NitroxModel/Discovery/GameInstallationFinder.cs [hunk 1]
```diff
@@ -13,8 +13,6 @@ public class GameInstallationFinder : IFindGameInstallation
     {
         private readonly List<IFindGameInstallation> finders = new List<IFindGameInstallation>
         {
-            new SteamGameRegistryFinder(),
-            new EpicGamesRegistryFinder(),
```
Comment (Sunrunner37) [inline NitroxModel/Discovery/GameInstallationFinder.cs]: Why were these removed?

File: NitroxClient/GameLogic/Vehicles.cs [hunk 2]
```diff
@@ -123,19 +123,54 @@ private void OnVehiclePrefabLoaded(GameObject prefab, string guid, Vector3 spawn
             Rigidbody rigidBody = gameObject.GetComponent<Rigidbody>();
             rigidBody.isKinematic = false;
             GuidHelper.SetNewGuid(gameObject, guid);
-            if (interactiveChildIdentifiers.IsPresent())
+
+            // Updates names and colours with persisted data .....yeah.....
+            if (CraftData.GetTechType(gameObject) != TechType.None)
```
Comment (Sunrunner37) [inline NitroxClient/GameLogic/Vehicles.cs]: Maybe pass in tech type so you can do `if(techType == TechType.Cyclops)`

File: NitroxClient/GameLogic/Constructor.cs [hunk 3]
```diff
@@ -35,8 +36,40 @@ public void BeginCrafting(GameObject constructor, TechType techType, float durat
...
                 string constructedObjectGuid = GuidHelper.GetGuid(constructedObject);
-                ConstructorBeginCrafting beginCrafting = new ConstructorBeginCrafting(constructorGuid, constructedObjectGuid, techType, duration, childIdentifiers, constructedObject.transform.position, constructedObject.transform.rotation);
+               
+                //Initialize some default values to serialize.
+                Vector3[] HSB = new Vector3[5];
+                Vector3[] Colours = new Vector3[5];
+                Vector4 tmpColour = Color.white;
+                string name = "Cyclops"; // Cant find a way to actually get the Cyclops name.
+
+                if (!vehicle)
```
Comment (Sunrunner37) [inline NitroxClient/GameLogic/Constructor.cs]: maybe break this into a new function such as `getVehicleAttributes(vehicle, out hsb, out colours, out name)`

File: NitroxModel/Discovery/GameInstallationFinder.cs (hunk 1 above)
Comment (ASimpson87) [reply NitroxModel/Discovery/GameInstallationFinder.cs]: uhh sorry that shouldn't be there... i have to do that as it causes problems since i have a few different installs. Removing.

**Comment by ASimpson87** [comment]
Added getObjectAttributes function to constructor , parsed techType to prefebloaded. Added Steam and Epic game finder back to gameinstallationfinder.
